apps/dotcom/client/src/tla/app/TldrawApp.ts

```diff
@@ -271,8 +277,19 @@ export class TldrawApp {
 		return { newFile, editorStoreSnapshot }
 	}
 
-	async deleteFile(_fileId: TldrawAppFileId) {
-		this.store.remove([_fileId])
+	async deleteOrForgetFile(fileId: TldrawAppFileId) {
+		if (this.isOwner(fileId)) {
+			this.store.remove([fileId])
+		} else {
+			const myId = this.getCurrentUserId()
+			const fileVisits = this.getAll('file-visit')
```

Comment:
i'm just realizing that this will grow prohibitively expensive as this list starts to include every visit x every user on the site, right?
any mitigations here to prevent this? (i see we do this in `getUserSharedFiles` as well)

Reply:
scaling for this will be solved when we eventually switch to a real db. and anyway I'm about to remove this record type.

Reply:
i can't tell what's fake and what's real anymore :insert-some-matrix-gif: